src/plugins/intel_gpu/src/graph/impls/ocl/gemm.cpp

```diff
@@ -154,6 +154,15 @@ struct gemm_impl : multi_stage_primitive<gemm> {
     }
 
     event::ptr execute_impl(const std::vector<event::ptr>& events, gemm_inst& instance) override {
+        if (instance.get_input_layout(0).count() == 0 ||
+            instance.get_input_layout(1).count() == 0) {
+            stream& stream = instance.get_network().get_stream();
+            for (const auto& ev : events) {
+                ev->wait();
```

Comment:
Why did you replace barrier with host-side synchronization? I think it's much worse from the performance standpoint

Reply:
Reverted to barrier solution